statsmodels/imputation/ros.py

```diff
@@ -0,0 +1,558 @@
+        A new dataframe with two new columns: "estimated" and "final".
+        The "estimated" column contains of the values inferred from the
+        best-fit line. The "final" column contains the estimated values
+        only where the original results were censored, and the original
+        results everwhere else.
+
+    """
+
+    # detect/non-detect selectors
+    uncensored_mask = df[censorship] == False
+    censored_mask = df[censorship] == True
+
+    # fit a line to the logs of the detected data
+    fit_params = stats.linregress(
+        df['Zprelim'][uncensored_mask],
+        transform_in(df[result][uncensored_mask])
+    )
+
+    # pull out the slope and intercept for use later
+    slope, intercept = fit_params[:2]
+
+    # model the data based on the best-fit curve
+    df = (
+        df.assign(estimated=transform_out(slope * df['Zprelim'][censored_mask] + intercept))
+          .assign(final=lambda df: numpy.where(df[censorship], df['estimated'], df[result]))
+    )
+
+    return df
+
+
+def _do_ros(df, result, censorship, transform_in, transform_out):
+    """
+    Prepares a dataframe for, and then esimates the values of a censored
+    dataset using Regression on Order Statistics
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+
+    result : str
+        Name of the column in the dataframe that contains observed
+        values. Censored values should be set to the detection (upper)
+        limit.
+
+    censorship : str
+        Name of the column in the dataframe that indicates that a
+        result is left-censored. (i.e., True -> censored,
+        False -> uncensored)
+
+    transform_in, transform_out : callable
+        Transformations to be applied to the data prior to fitting
+        the line and after estimated values from that line. Typically,
+        `numpy.log` and `numpy.exp` are used, respectively.
+
+    Returns
+    -------
+    estimated : pandas.DataFrame
+        A new dataframe with two new columns: "estimated" and "final".
+        The "estimated" column contains of the values inferred from the
+        best-fit line. The "final" column contains the estimated values
+        only where the original results were censored, and the original
+        results everwhere else.
+
+    """
+
+    # compute the Cohn numbers
+    cohn = cohn_numbers(df, result=result, censorship=censorship)
+
+    modeled = (
+        df.pipe(_ros_sort, result=result, censorship=censorship)
+          .assign(det_limit_index=lambda df: df[result].apply(_detection_limit_index, args=(cohn,)))
+          .assign(rank=lambda df: _ros_group_rank(df, 'det_limit_index', censorship))
+          .assign(plot_pos=lambda df: plotting_positions(df, censorship, cohn))
+          .assign(Zprelim=lambda df: stats.norm.ppf(df['plot_pos']))
+          .pipe(_ros_estimate, result, censorship, transform_in, transform_out)
+    )
+
+    return modeled
+
+
+def ROS(result, censorship, df=None, min_uncensored=2,
```

Comment:
redundant part of names depends on where we add it in the api.
If we add it at a higher level api directly, then `ros` might not be part of the path. If it's always accessed through `ros.xxx`, then an extra ros would not be very informative.

another problem: on Windows the file paths are not case sensitive, so sphinx cannot create html files for functions or classes that differ only in case. (we have a few problem cases like that in genmod links)